core/templates/services/math-interactions.service.ts

```diff
@@ -413,25 +413,31 @@ export class MathInteractionsService {
   }
 
   expressionMatchWithPlaceholders(
-      expressionWithPlaceholders: string, expressionWithoutPlaceholders: string,
+      expressionWithPlaceholders: string, inputExpression: string,
       placeholders: string[]): boolean {
-    // Check if expressionWithoutPlaceholders contains any placeholders.
-    for (let variable of nerdamer(expressionWithoutPlaceholders).variables()) {
+    // Check if inputExpression contains any placeholders.
+    for (let variable of nerdamer(inputExpression).variables()) {
       if (placeholders.includes(variable)) {
         return false;
       }
     }
 
     // The expressions are first split into terms by addition and subtraction.
     let termsWithPlaceholders = this.getTerms(expressionWithPlaceholders);
-    let termsWithoutPlaceholders = this.getTerms(expressionWithoutPlaceholders);
+    let inputTerms = this.getTerms(inputExpression);
 
+    // Each term in the expression containing placeholders is compared with
+    // terms in the input expression. Two terms are said to be matched iff
+    // upon subtracting or dividing them, the resultant contains only
+    // placeholders. This would imply that the input term matches with the
+    // general form(the term with placeholders).
```

Comment:
Oh, how does this work, so if we are comparing 2x and ax where a is placeholder, how would subtracting or dividing get rid of the 2?

Reply:
Ok, so when we divide 2x and ax, the resultant would be 2/a which contains only placeholder variables and that is why it is said to have matched. So things like 2x^2, 2y, etc. wouldn't match since the resultant would contain non-placeholder variables as well.

Reply:
Ah ok, so we are only checking for presence and absence. Yeah, looks good. Make sure that there are comprehensive tests for this.